--- channelcmd.template.cgi.GIT	2020-12-27 01:18:32.476485041 -0800
+++ channelcmd.template.cgi.WITH3HOUR_BLOCK_GUIDE_DATA_INTEGRATED	2020-12-11 22:57:15.832238951 -0800
@@ -93,26 +80,101 @@
 #
 
 
-#use date -d $XYZ + 30minutes or / 30min ago ??
-#
-#full date/time plus offset
-
 ZZSS=00:00
 TZSS=30:00
 FNOWUTC=$(date --utc +%FT%T%Z)
 FNOW=$(date -d "$(date +%FT%T) 480min" +"%FT%T")
 CFNOW=$(echo $FNOW|cut -c1-14)
-
 if [ `echo $FNOW|cut -d: -f2` -lt 30 ];
 	then
 	    HNOW=$(echo $CFNOW$ZZSS)
-	    echo "Current Time on Hour < 30, using $ZZSS/TIMESTAMP $HNOW for starttime query specifier"
+	    #echo "Current Time on Hour < 30, using $ZZSS/TIMESTAMP $HNOW for starttime query specifier"
 	else
 	    HNOW=$(echo $CFNOW$TZSS)
-	    echo "Current Time on Hour  > 30, using $TZSS/TIMESTAMP $HNOW for starttime query specifier"
+	    #echo "Current Time on Hour  > 30, using $TZSS/TIMESTAMP $HNOW for starttime query specifier"
+fi
+
+
+
+FNOWPREV30=$(date -d "$(date +%FT%T) 450min" +"%FT%T")
+CFNOWPREV30=$(echo $FNOWPREV30|cut -c1-14)
+if [ `echo $FNOWPREV30|cut -d: -f2` -lt 30 ];
+	then
+	    HNOWPREV30=$(echo $CFNOWPREV30$ZZSS)
+	    #echo "Current Time on Hour < 30, using $ZZSS/TIMESTAMP $HNOWPREV30 for starttime query specifier"
+	else
+	    HNOWPREV30=$(echo $CFNOWPREV30$TZSS)
+	    #echo "Current Time on Hour  > 30, using $TZSS/TIMESTAMP $HNOWPREV30 for starttime query specifier"
+fi
+
+
+FNOWPREV60=$(date -d "$(date +%FT%T) 420min" +"%FT%T")
+CFNOWPREV60=$(echo $FNOWPREV60|cut -c1-14)
+if [ `echo $FNOWPREV60|cut -d: -f2` -lt 30 ];
+	then
+	    HNOWPREV60=$(echo $CFNOWPREV60$ZZSS)
+	    #echo "Current Time on Hour < 30, using $ZZSS/TIMESTAMP $HNOWPRE60 for starttime query specifier"
+	else
+	    HNOWPREV60=$(echo $CFNOWPREV60$TZSS)
+	    #echo "Current Time on Hour  > 30, using $TZSS/TIMESTAMP $HNOWPREV60 for starttime query specifier"
+fi
+
+
+FNOWPREV90=$(date -d "$(date +%FT%T) 390min" +"%FT%T")
+CFNOWPREV90=$(echo $FNOWPREV90|cut -c1-14)
+if [ `echo $FNOWPREV90|cut -d: -f2` -lt 30 ];
+	then
+	    HNOWPREV90=$(echo $CFNOWPREV90$ZZSS)
+	    #echo "Current Time on Hour < 30, using $ZZSS/TIMESTAMP $HNOWPRE90 for starttime query specifier"
+	else
+	    HNOWPREV90=$(echo $CFNOWPREV90$TZSS)
+	    #echo "Current Time on Hour  > 30, using $TZSS/TIMESTAMP $HNOWPREV90 for starttime query specifier"
+fi
+
+
+FNOWPREV120=$(date -d "$(date +%FT%T) 360min" +"%FT%T")
+CFNOWPRE120=$(echo $FNOWPREV120|cut -c1-14)
+if [ `echo $FNOWPREV90|cut -d: -f2` -lt 30 ];
+	then
+	    HNOWPREV120=$(echo $CFNOWPREV120$ZZSS)
+	    #echo "Current Time on Hour < 30, using $ZZSS/TIMESTAMP $HNOWPRE120 for starttime query specifier"
+	else
+	    HNOWPREV120=$(echo $CFNOWPREV120$TZSS)
+	    #echo "Current Time on Hour  > 30, using $TZSS/TIMESTAMP $HNOWPREV120 for starttime query specifier"
+fi
+
+
+
+
+FNOW30=$(date -d "$(date +%FT%T) 510min" +"%FT%T")
+CFNOW30=$(echo $FNOW30|cut -c1-14)
+if [ `echo $FNOW30|cut -d: -f2` -lt 30 ];
+	then
+	    HNOW30=$(echo $CFNOW30$ZZSS)
+	    #echo "Current Time on Hour < 30, using $ZZSS/TIMESTAMP $HNOW30 for starttime query specifier"
+	else
+	    HNOW30=$(echo $CFNOW30$TZSS)
+	    #echo "Current Time on Hour  > 30, using $TZSS/TIMESTAMP $HNOW30 for starttime query specifier"
+fi
+
+
+
+FNOW60=$(date -d "$(date +%FT%T) 540min" +"%FT%T")
+CFNOW60=$(echo $FNOW60|cut -c1-14)
+if [ `echo $FNOW60|cut -d: -f2` -lt 30 ];
+	then
+	    HNOW60=$(echo $CFNOW60$ZZSS)
+	    #echo "Current Time on Hour < 30, using $ZZSS/TIMESTAMP $HNOW60 for starttime query specifier"
+	else
+	    HNOW60=$(echo $CFNOW60$TZSS)
+	    #echo "Current Time on Hour  > 30, using $TZSS/TIMESTAMP $HNOW60 for starttime query specifier"
 fi
 
 
+
+
+
+
 #echo $HNOW
 #echo $FNOW
 echo "<br>"
@@ -139,7 +201,8 @@
 MYTHSOURCEID=1
 MCHANNEL=$MCHANNEL
 CHANID="$MYTHSOURCEID$MCHANNEL"
-echo "mythdb Channelid(CHANID)=$CHANID"
+echo "<pre>Working on Mythtv's mythconverg database Channel(CHANID)=$CHANID</pre>"
+echo "<pre>*ALL Mythtv's mythconverg database times are in UTC</pre>"
 
 TITLE="//Title"
 SUBTITLE="//SubTitle"
@@ -148,23 +211,57 @@
 ENDTIME="//EndTime"
 
 
-echo "<pre>The Current iso8601 Local Time is:</pre>"
-echo "<pre>           $(date +%FT%T%Z)</pre>"
+echo "<pre>Local Time is: $(date +%FT%T%Z)</pre>"
+echo "<pre>UTC Time is:   $(echo $FNOWUTC)</pre>"
 
-echo "<pre>The Current UTC iso8601 Time is:</pre>"
-echo "<pre>           $(echo $FNOWUTC)</pre>"
+#echo "<h4>Previous 120minutes : </h4>"
+#echo "<pre>$(echo $HNOWPREV120)</pre>"
 
 
+echo "<pre>$(curl -s http://"$MYTHTV_SERVER:6544/Guide/GetProgramDetails?StartTime=$HNOWPREV120&ChanId=$CHANID" 2>&1|xmlstarlet sel -t -v $TITLE -nl  -v $SUBTITLE -nl -v $AIRDATE -nl -v $STARTTIME -nl -v $ENDTIME -nl)</pre>"
+
+
+#echo "<h4>Previous 90minutes : </h4>"
+#echo "<pre>$(echo $HNOWPREV90)</pre>"
+
+echo "<pre>$(curl -s http://"$MYTHTV_SERVER:6544/Guide/GetProgramDetails?StartTime=$HNOWPREV90&ChanId=$CHANID" 2>&1|xmlstarlet sel -t -v $TITLE -nl  -v $SUBTITLE -nl -v $AIRDATE -nl -v $STARTTIME -nl -v $ENDTIME -nl)</pre>"
+
+
+#echo "<h4>Previous 60minutes : </h4>"
+#echo "<pre>$(echo $HNOWPREV60)</pre>"
+
+echo "<pre>$(curl -s http://"$MYTHTV_SERVER:6544/Guide/GetProgramDetails?StartTime=$HNOWPREV60&ChanId=$CHANID" 2>&1|xmlstarlet sel -t -v $TITLE -nl  -v $SUBTITLE -nl -v $AIRDATE -nl -v $STARTTIME -nl -v $ENDTIME -nl)</pre>"
+
+
+#echo "<h4>Previous 30minutes : </h4>"
+#echo "<pre>$(echo $HNOWPREV30)</pre>"
+
+echo "<pre>$(curl -s http://"$MYTHTV_SERVER:6544/Guide/GetProgramDetails?StartTime=$HNOWPREV30&ChanId=$CHANID" 2>&1|xmlstarlet sel -t -v $TITLE -nl  -v $SUBTITLE -nl -v $AIRDATE -nl -v $STARTTIME -nl -v $ENDTIME -nl)</pre>"
+
+
+#echo "<h4>Current : </h4>"
+#echo "<pre>$(echo $HNOW)</pre>"
+
+echo "<pre>$(curl -s http://"$MYTHTV_SERVER:6544/Guide/GetProgramDetails?StartTime=$HNOW&ChanId=$CHANID" 2>&1|xmlstarlet sel -t -v $TITLE -nl -v $SUBTITLE -nl -v $AIRDATE -nl -v $STARTTIME -nl -v $ENDTIME -nl)</pre>"
+
+
+#echo "<h4>+30minutes : </h4>"
+#echo "<pre>$(echo $HNOW30)</pre>"
+
+echo "<pre>$(curl -s http://"$MYTHTV_SERVER:6544/Guide/GetProgramDetails?StartTime=$HNOW30&ChanId=$CHANID" 2>&1|xmlstarlet sel -t -v $TITLE -nl -v $SUBTITLE -nl -v $AIRDATE -nl -v $STARTTIME -nl -v $ENDTIME -nl)</pre>"
+
+
+#echo "<h4>+60minutes : </h4>"
+#echo "<pre>$(echo $HNOW60)</pre>"
+
+echo "<pre>$(curl -s http://"$MYTHTV_SERVER:6544/Guide/GetProgramDetails?StartTime=$HNOW60&ChanId=$CHANID" 2>&1|xmlstarlet sel -t -v $TITLE -nl -v $SUBTITLE -nl -v $AIRDATE -nl -v $STARTTIME -nl -v $ENDTIME -nl)</pre>"
+
+#echo "<pre>$(curl -s http://"$MYTHTV_SERVER:6544/Guide/GetProgramDetails?StartTime=$HNOW60&ChanId=$CHANID" 2>&1|xmlstarlet sel -t -o "Title: " -v $TITLE -nl -o "        SubTitle: " -v $SUBTITLE -nl -o "         Orig. Airdate: " -v $AIRDATE -nl -nl -o "StartTime: " -v $STARTTIME -nl -o "  EndTime: " -v $ENDTIME -nl)</pre>"
+
 
 
-echo "<h4>Current Program: </h4>"
 
-echo "<pre>$(curl -s http://"$MYTHTV_SERVER:6544/Guide/GetProgramDetails?StartTime=$HNOW&ChanId=$CHANID" 2>&1|xmlstarlet sel -t -o "Title: " -v $TITLE -nl -o "        SubTitle: " -v $SUBTITLE -nl -o "         Orig. Airdate: " -v $AIRDATE -nl -nl -o "StartTime: " -v $STARTTIME -nl -o "  EndTime: " -v $ENDTIME -nl)</pre>"
 
-#echo "<h4>Next Program: </h4>"
-#echo "<pre>$(echo $PNEXTHR)</pre>"
-#
-#echo "<pre>$(curl -s http://"$MYTHTV_SERVER:6544/Guide/GetProgramDetails?StartTime=$PNEXTHR&ChanId=$CHANID" 2>&1|xmlstarlet sel -t -o "Title: " -v $TITLE -nl -o "        SubTitle: " -v $SUBTITLE -nl -o "         Orig. Airdate: " -v $AIRDATE -nl -nl -o "StartTime: " -v $STARTTIME -nl -o "  EndTime: " -v $ENDTIME -nl)</pre>"
 
